### DISCUSSION TITLE
Metrics aggregation pipeline

### DISCUSSION THREAD

**Discussion by bogdandrutu** [open-telemetry/opentelemetry-java#2378] [Q&A]
**Here is the picture that describes the MetricsSDK design:**

![MetricsSDK](https://user-images.githubusercontent.com/1373887/102817019-d21fac80-4383-11eb-827c-8620ce54414d.jpg)

Some details:
* Every instrument has associated (based on the instrument type) an `Accumulator`.
* The `Accumulator` passes all the accumulated pairs <`LabelSet`, `Accumulation`> to the `Processor`.
* `Accumulation` is an immutable representation of an accumulated value by the `Aggregator` in case of a synchronous instrument, or created by the asynchronous accumulator in case of async instruments.

**Comment by bogdandrutu** [comment]
@jkwatson I will update the picture with detail in the processor once I get to work and clear that code. For the moment I am only focused on the aggregator/accumulation stuff.

**Reply by jkwatson** [reply]
awesome. sounds good. this helps my visual brain quite a bit.  👍
